From d3f1371a1e19899331d73732d1c6abd71054918b Mon Sep 17 00:00:00 2001
From: Benjamin Baratte <benjamin.baratte@st.com>
Date: Thu, 5 Mar 2026 13:57:18 +0000
Subject: [PATCH 04/14] Allow SE key to use key vendor id within PSA crypto

Signed-off-by: Benjamin Baratte <benjamin.baratte@st.com>
---
 core/psa_crypto.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/psa_crypto.c b/core/psa_crypto.c
index de131f5..5d09f05 100644
--- a/core/psa_crypto.c
+++ b/core/psa_crypto.c
@@ -1765,7 +1765,11 @@ static psa_status_t psa_validate_key_attributes(const psa_key_attributes_t *attr
             return PSA_ERROR_INVALID_ARGUMENT;
         }
     } else {
+#ifdef MBEDTLS_PSA_CRYPTO_SE_C
+        if (!psa_is_valid_key_id(psa_get_key_id(attributes), 1)) {
+#else
         if (!psa_is_valid_key_id(psa_get_key_id(attributes), 0)) {
+#endif
             return PSA_ERROR_INVALID_ARGUMENT;
         }
     }
-- 
2.43.0

