From 09480d0cd5ec03ca1f54b3fb10a1845b10357655 Mon Sep 17 00:00:00 2001
From: David Vincze <david.vincze@arm.com>
Date: Thu, 12 Mar 2026 13:41:43 +0000
Subject: [PATCH 12/16] Exclude context definitions for PSA Crypto client

Signed-off-by: David Vincze <david.vincze@arm.com>
---
 include/psa/crypto_struct.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/psa/crypto_struct.h b/include/psa/crypto_struct.h
index 0cc32e7..1eab375 100644
--- a/include/psa/crypto_struct.h
+++ b/include/psa/crypto_struct.h
@@ -56,9 +56,11 @@ extern "C" {
 
 #include "tf-psa-crypto/build_info.h"
 
+#if defined(MBEDTLS_PSA_CRYPTO_C)
 /* Include the context definition for the compiled-in drivers for the primitive
  * algorithms. */
 #include "psa/crypto_driver_contexts_primitives.h"
+#endif
 
 struct psa_hash_operation_s {
 #if defined(MBEDTLS_PSA_CRYPTO_CLIENT) && !defined(MBEDTLS_PSA_CRYPTO_C)
@@ -117,9 +119,11 @@ static inline struct psa_cipher_operation_s psa_cipher_operation_init(void)
     return v;
 }
 
+#if defined(MBEDTLS_PSA_CRYPTO_C)
 /* Include the context definition for the compiled-in drivers for the composite
  * algorithms. */
 #include "psa/crypto_driver_contexts_composites.h"
+#endif
 
 struct psa_mac_operation_s {
 #if defined(MBEDTLS_PSA_CRYPTO_CLIENT) && !defined(MBEDTLS_PSA_CRYPTO_C)
@@ -188,9 +192,11 @@ static inline struct psa_aead_operation_s psa_aead_operation_init(void)
     return v;
 }
 
+#if defined(MBEDTLS_PSA_CRYPTO_C)
 /* Include the context definition for the compiled-in drivers for the key
  * derivation algorithms. */
 #include "psa/crypto_driver_contexts_key_derivation.h"
+#endif
 
 struct psa_key_derivation_s {
 #if defined(MBEDTLS_PSA_CRYPTO_CLIENT) && !defined(MBEDTLS_PSA_CRYPTO_C)
-- 
2.43.0

